# Bonus: Temperature Analysis I

In [1]:
import pandas as pd
from datetime import datetime as dt

In [2]:
# "tobs" is "temperature observations"
df = pd.read_csv('Resources/hawaii_measurements.csv')
df.head()

station        date  prcp  tobs
0  USC00519397  2010-01-01  0.08    65
1  USC00519397  2010-01-02  0.00    63
2  USC00519397  2010-01-03  0.00    74
3  USC00519397  2010-01-04  0.00    76
4  USC00519397  2010-01-06   NaN    73

In [3]:
# Convert the date column format from string to datetime
df['date'] = pd.to_datetime(df['date'], format='%Y-%m-%d')

In [4]:
# Set the date column as the DataFrame index
df = df.set_index('date')

### Compare June and December data across all years 

In [5]:
from scipy import stats

In [6]:
# Preview data frame
df.head()

station  prcp  tobs
date                               
2010-01-01  USC00519397  0.08    65
2010-01-02  USC00519397  0.00    63
2010-01-03  USC00519397  0.00    74
2010-01-04  USC00519397  0.00    76
2010-01-06  USC00519397   NaN    73

In [9]:
# Filter data for desired months
june_df = df[df.index.month == 6]
dec_df = df[df.index.month == 12]

In [10]:
# Identify the average temperature for June
average_june = june_df['tobs'].mean()
average_june

74.94411764705882

In [11]:
# Identify the average temperature for December
average_dec = dec_df['tobs'].mean()
average_dec

71.04152933421226

In [18]:
difference = average_june - average_dec
difference

3.902588312846561

In [12]:
# Create collections of temperature data
june_temps = june_df['tobs']
dec_temps = dec_df['tobs']

# Preview data
june_temps

date
2010-06-01    78
2010-06-02    76
2010-06-03    78
2010-06-04    76
2010-06-05    77
              ..
2017-06-26    79
2017-06-27    74
2017-06-28    74
2017-06-29    76
2017-06-30    75
Name: tobs, Length: 1700, dtype: int64

In [16]:
dec_temps

date
2010-12-01    76
2010-12-03    74
2010-12-04    74
2010-12-06    64
2010-12-07    64
              ..
2016-12-27    71
2016-12-28    71
2016-12-29    69
2016-12-30    65
2016-12-31    65
Name: tobs, Length: 1517, dtype: int64

In [19]:
print(dec_temps.describe())

count    1517.000000
mean       71.041529
std         3.745920
min        56.000000
25%        69.000000
50%        71.000000
75%        74.000000
max        83.000000
Name: tobs, dtype: float64


In [20]:
print(june_temps.describe())

count    1700.000000
mean       74.944118
std         3.257417
min        64.000000
25%        73.000000
50%        75.000000
75%        77.000000
max        85.000000
Name: tobs, dtype: float64


In [17]:
# Run t-test
stats.ttest_ind(june_temps, dec_temps)

Ttest_indResult(statistic=31.60372399000329, pvalue=3.9025129038616655e-191)

### Analysis

The t-test shows that the p-value is an extremely small number so there is very strong evidence that it rejects the null hypothesis that the two groups are equal. The difference between June and December's average temps is not much—3.90 degrees!